### PR TITLE
tests: include Elasticsearch 5 in Travis-CI 

### PR DESCRIPTION
**Pull request by hjhsalo**
* include Elasticsearch5 in Travis-CI tests.

Signed-off-by: Harri Hirvonsalo <harri.hirvonsalo@cern.ch>

### CODE REVIEW

**Comment by hjhsalo** [comment]
ES5 tests in Travis are failing but I think otherwise the tasks defined in #103 are done, so I guess this is ready for review?

If the reviewer has knowledge about `.travis.yml`, could you suggest a way to reduce the repetion of `ES_DOWNLOAD_URL=https://download.elasticsearch.org/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/${ES_VERSION}/elasticsearch-${ES_VERSION}.tar.gz` to something like `ES_DOWNLOAD_URL=ES_2_DOWNLOAD_URL`? The same also applies to download URL of ES5.
I couldn't come up with solutions by myself...

**Comment by hjhsalo** [comment]
Please note that files https://github.com/inveniosoftware/invenio/blob/master/scripts/provision-elasticsearch.sh and https://github.com/inveniosoftware/invenio/blob/master/elasticsearch/Dockerfile should probably be prepared for ES5.

Maybe something like this could be used in `Dockerfile` and then change behaviour of `provision-elasticsearch.sh` based on value of `$ES_VERSION`:
```
ARG ES_VERSION=2
ENV ES_VERSION ${ES_VERSION:-2}

FROM elasticsearch:$ES_VERSION
...
```

Also note that *mapper-attachment* -plugin is deprecated in ES5 in favor of *ingest-attachment* -plugin and that name of the binary that is used to install plugins changes in ES5 as well (`$ elasticsearch-plugin install ...` VS `$ plugin install ...`)

https://www.elastic.co/guide/en/elasticsearch/plugins/5.6/mapper-attachments.html
https://www.elastic.co/guide/en/elasticsearch/plugins/5.6/ingest-attachment.html

**Comment by nharraud** [comment]
The current state of this PR does not enable to support elasticsearch 5.
* `index init` fails because the mappings provided by invenio-marc21 are not elasticsearch 5 compatible (see the [breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/breaking_50_mapping_changes.html))
* The percolator API changed and breaks the tests (see the [breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/breaking_50_percolator.html))

Currently invenio-search uses all the mappings provided. However  if we want to support both elasticsearch-5 and elasticsearch-2 we would need to provide two different set of mappings. Thus we would need to refactor invenio-search.

@inveniosoftware/architects Another issue pointed out by @slint : [**Elasticsearch version 2 End Of Life is 2018-02-28**](https://www.elastic.co/support/eol). We might want to go for version 5 or 6 immediately and drop version 2.

**Comment by slint** [comment]
There's also an [official migration helper plugin](https://github.com/elastic/elasticsearch-migration/tree/2.x) that goes over some basic checks of the cluster. I've added the output of running the tool on the [relevant issue #103](https://github.com/inveniosoftware/invenio-oaiserver/issues/103#issuecomment-346672967)

**Comment by slint** [comment]
Dependencies (PR merged and released):

- inveniosoftware/invenio-marc21#56
    - inveniosoftware/invenio-records-rest/issues/173
- inveniosoftware/invenio-search/pull/119

File: invenio_oaiserver/percolator.py [hunk 1]
```diff
@@ -26,21 +26,33 @@
...
+from elasticsearch import VERSION as ES_VERSION
 from invenio_indexer.api import RecordIndexer
 from invenio_search import current_search, current_search_client
 
 from .models import OAISet
 from .proxies import current_oaiserver
 from .query import query_string_parser
 
+# ES5 percolator
+PERCROLATOR_DOC_TYPE = '.percolators' if ES_VERSION[0] == 2 else 'percolators'
```
Comment (krzysztof) [inline invenio_oaiserver/percolator.py]: ``PERCROLATOR`` -> ``PERCOLATOR``

**Pull request merged**